pkg/kv/kvserver/replica_application_result.go

```diff
@@ -501,26 +501,29 @@ func (r *Replica) handleLeaseResult(
 // stagePendingTruncationRaftMuLocked installs the new RaftTruncatedState,
 // updates the log size, and truncates the raft log cache.
 func (r *Replica) stagePendingTruncationRaftMuLocked(pt pendingTruncation) {
+	r.asLogStorage().stagePendingTruncationRaftMuLocked(pt)
+}
+
+func (r *replicaLogStorage) stagePendingTruncationRaftMuLocked(pt pendingTruncation) {
```

Comment:
Move this to `replica_raftlog_writes.go`, as well as other truncation-related code. For followups.